sqlmesh/core/state_sync/common.py

```diff
@@ -33,13 +34,20 @@ def cleanup_expired_views(adapter: EngineAdapter, environments: t.List[Environme
     expired_table_environments = [
         environment for environment in environments if environment.suffix_target.is_table
     ]
-    for expired_schema in {
-        snapshot.qualified_view_name.schema_for_environment(environment.naming_info)
+    for expired_catalog, expired_schema in {
+        (
+            snapshot.qualified_view_name.catalog,
+            snapshot.qualified_view_name.schema_for_environment(environment.naming_info),
+        )
         for environment in expired_schema_environments
         for snapshot in environment.snapshots
         if snapshot.is_model
     }:
-        adapter.drop_schema(expired_schema, ignore_if_not_exists=True, cascade=True)
+        adapter.drop_schema(
+            nullsafe_join(".", expired_catalog, expired_schema),
```

Comment:
schema_(catalog, schema)

Reply:
I think the qualified view stuff currently has a bug. We parse a snapshot name into a table (that part is fine) but then get the string representation of the identifiers instead of the identifiers themsevles: https://github.com/TobikoData/sqlmesh/blob/f8e45bbc20f5dda759728c260bc194dff8d536ce/sqlmesh/core/snapshot/definition.py#L215-L220

Since we convert to string, I think doing a string join here is correct. If we had identifiers or potentially strings that are properly normalized then we would do `schemo_`. At least that is how I think it is supposed to work based on `to_table` and `table_`.

Reply:
hm not sure i understand, let's chat about this